src/riak_kv_qry_compiler.erl

```diff
@@ -641,6 +663,7 @@ break_out_timeseries(Filters1, LocalFields1, [QuantumFields]) ->
         {_, {{_,_,{_,Starts}}, {_,_,{_,Ends}}}} when is_integer(Starts),
                                                      is_integer(Ends),
                                                      Starts > Ends ->
+            %% FIXME reliance on three element key!?
```

Comment:
Realised afterwards that this is matching AST which is also three elements, added a test to be sure.